app/services/stories_api/v3/endpoints/story_item.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+module StoriesApi
+  module V3
+    module Endpoints
+      class StoryItem
+        include Helpers
+
+        attr_reader :params, :user, :story, :item
+
+        def initialize(params)
+          @params = params
+          @user = SupplejackApi::User.find_by_api_key(params[:api_key])
+          @story = @user ? @user.user_sets.find_by_id(params[:story_id]) : nil
+          @item = @story ? @story.set_items.find_by_id(params[:id]) : nil
+        end
+
+        def get
+          create_response(status: 200, payload: ::StoriesApi::V3::Presenters::StoryItem.new.call(item))
+        end
+
+        def patch
+          return create_exception('MandatoryParamMissing', param: :item) unless params[:item]
+
+          merge_patch = PerformMergePatch.new(::StoriesApi::V3::Schemas::StoryItem::BlockValidator.new,
+                                              ::StoriesApi::V3::Presenters::StoryItem.new)
+
+          valid = merge_patch.call(item, params[:item])
+
+          return create_exception('SchemaValidationError', errors: merge_patch.validation_errors) unless valid
+
+          item.save
+          create_response(status: 200, payload: ::StoriesApi::V3::Presenters::StoryItem.new.call(item))
+        end
+
+        def delete
+          item.delete
+
+          create_response(status: 204)
+        end
+
+        # Returns error if story and user were not initialised
+        #
+        # @author Eddie
+        # @last_modified Eddie
+        # @return [Hash] the error
+        def errors
+          return create_exception('UserNotFound', id: params[:api_key]) unless user.present?
+          return create_exception('StoryNotFound', id: params[:story_id]) unless story.present?
+          return create_exception('StoryItemNotFound',
+                                  item_id: params[:id],
+                                  story_id: params[:story_id]) unless item.present?
+        end
```

Comment:
This is very much a matter of opinion, but I find it a bit weird how there is a bunch of ternary logic around finding stuff in the `init` method, and then an unless chain here in errors. Surely when we can't find the user (for example) in the `init` we could add the error then and abort. I guess it feels like the order of initializations is captured in both places, redundantly.  I guess if an action method could fail with errors mid operation, then this error logic would become much more complicated.